src/evaluate_fences.rs

```diff
@@ -68,8 +68,19 @@ pub fn evaluate_fences<'fencecollectionlifetime, 'sourcefilelifetime>(
     source_files: &HashMap<String, SourceFile>,
     tsconfig_paths_json: &TsconfigPathsJson,
     source_file: &'sourcefilelifetime SourceFile,
+    ignored_dirs: Vec<String>,
 ) -> Result<Option<Vec<ImportRuleViolation<'fencecollectionlifetime, 'sourcefilelifetime>>>, String>
 {
+    match ignored_dirs.iter().find(|id| {
+        regex::Regex::new(&id.as_str())
+            .expect("unable to match --ignoredDirs")
```

Comment:
this error message should include the regexp that failed to compile

Reply:
This recompiles the regexp on every single fence. That is needlessly wasteful.

The regexps should be compiled once, and used to filter out ignored fences during walk_dirs.